### PR TITLE
Promote into methods

### PR DESCRIPTION
**Pull request by zancas**
@dangle 
This change promotes common functionality into `utils.py` by examining:

`containenv/containenv/cli/__init__.py`

You'll see what I've done...     by imagining:

`containenv/contain/cli/__init__.py`

I think it'll become apparent why I've done it!  :)

### CODE REVIEW

File: containenv/containenv/cli/__init__.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+Options:
+  -V, --version     Display the version and exit.
+  -v, --verbose     Print logging information to the console.
+  -d <LEVEL>, --debug <LEVEL>  Set the sensitivity of loggers to [default: 30]
+
+{}
+
+See 'containenv help COMMAND' for more information on a specific COMMAND.
+NOTE:  See also 'contain help' which is included in this package.
```
Comment (dangle) [inline containenv/containenv/cli/__init__.py:13]: Do they really need to know about `contain` here?

File: containenv/utils.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
+from __future__ import print_function
+
+try:
+    from cStringIO import StringIO
+except ImportError:
```
Comment (dangle) [inline containenv/utils.py:5]: Instead of manually checking this, I think we should use `six`.

File: containenv/utils.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler(sys.stderr))
+
+
+class ContainEnvDockerfileDoesNotExist(Exception):
+    pass
+
+
+class ContainEnvDockerfileAlreadyExists(Exception):
+    pass
```
Comment (dangle) [inline containenv/utils.py:33]: Pull these exceptions out into `exceptions.py`, maybe.

File: containenv/utils.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        docopt(doc, argv=('help',))
+            
+    
+    def _get_command(self, tokens):
+        commands = []
+        for key in tokens:
+            if not key.startswith('<') \
+            and not key.startswith('-') \
+            and not key.isupper():
+                '''See: https://github.com/docopt/docopt#usage-pattern-format'''
```
Comment (dangle) [inline containenv/utils.py:180]: Move this to the function docstring.

File: containenv/utils.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+            raise ValueError("Unrecognized option '{}'.".format(help_subject_cmd))
+        else:
+            mod = self._get_command_module(help_subject_cmd)
+            logging.debug("mod is {}".format(mod))
+            doc = mod.__doc__
+        logging.debug(doc)
+        docopt(doc, argv=('help',))
+            
+    
+    def _get_command(self, tokens):
```
Comment (dangle) [inline containenv/utils.py:174]: I still think the subcommand argument is the way to go, per the examples. This feels fragile.

**Review by dangle — APPROVED** [review]

File: containenv/containenv/cli/__init__.py (hunk 1 above)
Comment (zancas) [reply containenv/containenv/cli/__init__.py:13]: I dunno' is it TMI?

File: containenv/utils.py (hunk 2 above)
Comment (zancas) [reply containenv/utils.py:5]: I'll issue.

Comment (zancas) [reply containenv/utils.py:5]: https://github.com/zancas/containenv/issues/13

File: containenv/utils.py (hunk 3 above)
Comment (zancas) [reply containenv/utils.py:33]: https://github.com/zancas/containenv/issues/14

File: containenv/utils.py (hunk 4 above)
Comment (zancas) [reply containenv/utils.py:180]: Done upstream

File: containenv/utils.py (hunk 5 above)
Comment (zancas) [reply containenv/utils.py:174]: Can you create an issue for this.   I don't understand the difference yet.

**Pull request merged**